application/dash_application/utility/plot_builder.py

```diff
@@ -290,16 +347,39 @@ def generate(title_text, sorted_data, x_fn, y_fn, axis_text, colourby, shapeby,
             y=y_fn(highlight_df),
             name="Highlighted Samples",
             mode='markers',
+            legendgroup="gsi_reserved_highlighted_samples",
+            showlegend=showlegend,
             marker={
                 "symbol": highlight_df['shape'],
                 "color": highlight_df['colour'],
                 "size": highlight_df['markersize'],
                 "opacity": 1
             }
         ))
+
+    return traces
+
+
+# writing a factory may be peak Java poisoning but it might help with all these parameters
+def generate(title_text, sorted_data, x_fn, y_fn, axis_text, graph_params,
```

Comment:
Could this be renamed to explain better what it's generating? (vs generate_subplot, generate_traces, generate_graphs etc)